src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -7973,6 +7973,7 @@ instruct storeimmL0_volatile(immL0 zero, /* sync_memory*/indirect mem)
 instruct storeP_volatile(iRegP src, /* sync_memory*/indirect mem)
 %{
   match(Set mem (StoreP mem src));
+  predicate(n->as_Store()->barrier_data() == 0);
```

Comment:
Why does the instruction below (`storeimmP0_volatile`) not require a similar predicate? Just checking - I guess the omission is intentional since it follows from the ZGC repo.

Reply:
No, you are right - it does need a similar predicate. The storeimmP0_volatile rule seems to be quite recent, and I wrote this code before it existed and then we must have merged without noticing. Nice catch!

Reply:
Yeah, nice catch! This would have been a bad bug. SATB could have missed Oops which were overwritten by Store0.